frontend/src/models/archive.ts

```diff
@@ -0,0 +1,41 @@
+export interface Archive {
+    archiveID: number;
+    archiveURL: string;
+}
+
+export interface ArchiveJson {
+    archiveID: number;
+    archiveURL: string;
+}
+
+export function archiveFromJson(json: ArchiveJson): Archive {
+    const { archiveID, archiveURL } = json;
+    return {
+        archiveID,
+        archiveURL: lengthenArchiveURL(archiveURL),
+    };
+}
+
+export function archiveToJson(archive: Archive): ArchiveJson {
```

Comment:
What's this function for?

Reply:
I store each URL as the 11-character Youtube IDs, so there should be conversion before using. I agree it is a bit convoluted, but it helps when I finally write a script to import the archives.